File: benchmarks/file_io/file_io.py
import sys
import os
import tempfile

n = int(sys.argv[1]) if len(sys.argv) > 1 else 1000000

# Create temp file
fd, filename = tempfile.mkstemp(prefix='barbell_file_io_')
os.close(fd)

try:
    # Write n bytes
    data = bytes(i % 256 for i in range(n))
    with open(filename, 'wb') as f:
        f.write(data)

    # Read back and sum
    with open(filename, 'rb') as f:
        read_data = f.read()

    total = sum(read_data)
    print(total)
finally:
    os.unlink(filename)

File: benchmarks/file_io/file_io.rb
require 'tempfile'

n = ARGV[0] ? ARGV[0].to_i : 1000000

# Create temp file
file = Tempfile.new('barbell_file_io_')
filename = file.path
file.close

begin
  # Write n bytes
  data = (0...n).map { |i| (i % 256).chr }.join.b
  File.binwrite(filename, data)

  # Read back and sum
  read_data = File.binread(filename)
  total = read_data.bytes.sum

  puts total
ensure
  File.unlink(filename) if File.exist?(filename)
end
